<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>training_form</class>
 <widget class="QWidget" name="training_form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1055</width>
    <height>707</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="training_layout">
     <item>
      <layout class="QVBoxLayout" name="left_side_layout">
       <property name="bottomMargin">
        <number>10</number>
       </property>
       <item>
        <widget class="QGroupBox" name="train_state_wrapper">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="training_state">
            <property name="font">
             <font>
              <pointsize>14</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Unknown</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="model_architecture_line">
         <property name="topMargin">
          <number>15</number>
         </property>
         <item>
          <widget class="QLabel" name="model_architecture_title">
           <property name="text">
            <string>Model architecture</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="model_name_combobox">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>200</width>
             <height>0</height>
            </size>
           </property>
           <item>
            <property name="text">
             <string>YOLOv5-Nano</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <spacer name="model_architecture_line_spacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="training_config_line">
         <property name="bottomMargin">
          <number>15</number>
         </property>
         <item>
          <widget class="QLabel" name="epochs_title">
           <property name="text">
            <string>Epochs</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="num_epochs_spin">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>10000</number>
           </property>
           <property name="value">
            <number>80</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="batch_size_title">
           <property name="text">
            <string>Batch size</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="batch_size_spin">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>512</number>
           </property>
           <property name="value">
            <number>8</number>
           </property>
           <property name="displayIntegerBase">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="learning_rate_title">
           <property name="text">
            <string>Learning rate</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="learning_rate_spin">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="decimals">
            <number>5</number>
           </property>
           <property name="minimum">
            <double>0.000010000000000</double>
           </property>
           <property name="maximum">
            <double>0.010000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.000010000000000</double>
           </property>
           <property name="value">
            <double>0.001000000000000</double>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="training_config_line_spacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="train_button_layout">
         <property name="leftMargin">
          <number>6</number>
         </property>
         <property name="topMargin">
          <number>6</number>
         </property>
         <property name="rightMargin">
          <number>6</number>
         </property>
         <property name="bottomMargin">
          <number>20</number>
         </property>
         <item>
          <widget class="QPushButton" name="train_button">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="styleSheet">
            <string notr="true">font-size: 12pt;
padding-left: 20;</string>
           </property>
           <property name="text">
            <string>Train</string>
           </property>
           <property name="icon">
            <iconset resource="../../../resources/resources.qrc">
             <normaloff>:/images/images/run.png</normaloff>:/images/images/run.png</iconset>
           </property>
           <property name="iconSize">
            <size>
             <width>24</width>
             <height>24</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QWidget" name="widget" native="true"/>
       </item>
       <item>
        <widget class="QLabel" name="losses_title">
         <property name="text">
          <string>Losses</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PlotWidget" name="loss_graph" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>200</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="metrics_title">
         <property name="text">
          <string>Metrics</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="PlotWidget" name="metric_graph" native="true">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>200</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="train_config_spacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QWidget" name="widget_2" native="true"/>
       </item>
      </layout>
     </item>
     <item>
      <widget class="Line" name="vertical_line">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="right_side_layout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="system_monitor_title">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>System monitor</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="system_monitor_edit">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>100</height>
          </size>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="logging_title">
         <property name="text">
          <string>Logging</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="log_edit">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header location="global">pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../resources/resources.qrc"/>
 </resources>
 <connections/>
</ui>
